src/components/TagPicker/index.tsx

```diff
@@ -10,22 +10,64 @@ import * as OptionsListUtils from '@libs/OptionsListUtils';
 import * as PolicyUtils from '@libs/PolicyUtils';
 import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
-import {defaultProps, propTypes} from './tagPickerPropTypes';
+import type {PolicyTag, PolicyTagList, PolicyTags, RecentlyUsedTags} from '@src/types/onyx';
 
-function TagPicker({selectedTag, tag, tagIndex, policyTags, policyRecentlyUsedTags, shouldShowDisabledAndSelectedOption, insets, onSubmit}) {
+type SelectedTagOption = {
+    name: string;
+    enabled: boolean;
+    accountID: number | null;
+};
+
+type TagPickerOnyxProps = {
+    /** Collection of tags attached to a policy */
+    policyTags: OnyxEntry<PolicyTagList>;
+
+    /** List of recently used tags */
+    policyRecentlyUsedTags: OnyxEntry<RecentlyUsedTags>;
+};
+
+type TagPickerProps = TagPickerOnyxProps & {
+    /** The policyID we are getting tags for */
+    // It's used in withOnyx HOC.
+    // eslint-disable-next-line react/no-unused-prop-types
+    policyID: string;
+
+    /** The selected tag of the money request */
+    selectedTag: string;
+
+    /** The name of tag list we are getting tags for */
+    tagListName: string;
+
+    /** Callback to submit the selected tag */
+    onSubmit: () => void;
+
+    /**
+     * Safe area insets required for reflecting the portion of the view,
+     * that is not covered by navigation bars, tab bars, toolbars, and other ancestor views.
+     */
+    insets: EdgeInsets;
+
+    /** Should show the selected option that is disabled? */
+    shouldShowDisabledAndSelectedOption?: boolean;
+
+    /** The index of a tag list */
```

Comment:
This is also unclear, not sure what this is tbh

Reply:
this is an index when you have multilevel tags and its just an number which tag list we selected , so I think this description is saying exactly what it is

https://github.com/Expensify/App/assets/25230417/067974d4-4818-4443-ac5c-4e710974995c

Reply:
>I think this description is saying exactly what it is

I disagree. The description is repeating what the code is already saying, which is unclear.

So I would suggest:
```suggestion
    /** Indicates which tag list index was selected */
    tagListIndex: number;
```

Reply:
Sorry @kubabutkiewicz , looks like you missed my suggestion of renaming the var to `tagListIndex` 😬 (I now see it was not obvious)